CMakeLists.txt

```diff
@@ -2,51 +2,87 @@ project(libMCNP2CAD)
 
 cmake_minimum_required(VERSION 2.8)
 
-set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -DHAVE_IGEOM_CONE" )
+# Build options
+option(BUILD_MCNP2CAD_TESTS "Build mcnp2cad tests" ON)
 
-# adjust compiler setting for Linux using gcc version 5.0 and higher
-if(CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
-    if(NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
+# Default to a release build
+if (NOT CMAKE_BUILD_TYPE)
+  message(STATUS "CMAKE_BUILD_TYPE not specified, defaulting to Release")
+  set(CMAKE_BUILD_TYPE Release)
+endif ()
+  
+# Use C++11
+set(CMAKE_CXX_STANDARD 11)
+
+add_definitions(-DHAVE_IGEOM_CONE=ON)
+
+# Adjust compiler setting for Linux using gcc version 5.0 and higher
+if (CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
+    if (NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
       set(CMAKE_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 ${CMAKE_CXX_FLAGS}")
-    endif()
-  endif()
-endif()
-
-if(${BUILD_CLI})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/cli)
-endif()
-
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
```

Comment:
Don't we still need to find Eigen3

Reply:
Not as far as I can tell. It finds it automatically for me. I could put it back in just in case though.

Reply:
Build systems are supposed to defend against "it works for me" circumstances that developers experience.  Someone else may have a configuration where it is not found so easily....

Reply:
I looked into this some more. As far as I can tell, this line wasn't doing anything in the first place. You would need to add the line `include_directories(${EIGEN3_INCLUDE_DIRS})` for it to do anything.

The reason it worked despite this is that Eigen3 is located in `/usr/include`, and you don't need to specify `/usr/include` as a included directory because it's already included by default.

Reply:
Should we always assume that Eigen is installed in the include path?

Reply:
I missed this -- I'm going to re-introduce this line

Reply:
Adding this line back in gives an error saying that it can't be used until CMake 3.0, so this can't be added without removing support for CMake 2.8. I'm going to add the line anyway and change the required version since 3.0 is still relatively ancient

Reply:
@gonuke if you `find_package` eigen3 on a system with system eigen3, it sets `EIGEN3_INCLUDE_DIR` to `/usr/include/eigen3`. However, this directory is not useful because when we include eigen in the c++ files, we use the line `#include <eigen3/Eigen/Eigen>`. Should these lines be changed to `#include <Eigen/Eigen>`?

Reply:
That change is probably the right thing to do

Reply:
Just pushed a commit that made this change. It added these lines:
```
find_package(Eigen3 3.3 REQUIRED)
message(STATUS "EIGEN3_INCLUDE_DIR: ${EIGEN3_INCLUDE_DIR}")
include_directories(${EIGEN3_INCLUDE_DIR})
```